### PR TITLE
chore: enable working with node v16.x LTS and upgrade typescript

### PR DESCRIPTION
**Pull request by yukinoda**
# enable working with node v16.x LTS and upgrade typescript

## What are you adding?
- upgrade [node-sass](https://www.npmjs.com/package/node-sass) dependency to enable working with node v16.x (Current LTS version)
- upgrade typescript
  - typescript error appeared after doing `yarn start` with node v16.x
  - => upgrade typescript and fix type errors due to the upgrade
 
NOTE: Discussion needed from typescript upgrade https://github.com/monstar-lab-oss/admin-panel-template-reactjs/discussions/134

## Breaking changes?
- node-sass version: `4.14.1` => `6.0.1`
- typescript version: `4.1.3` => `4.5.2`

### CODE REVIEW

**Review by yukinoda — COMMENTED** [review]
Ready to review

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "i18next": "^20.3.2",
     "lodash": "^4.17.21",
     "moment": "^2.29.1",
-    "node-sass": "^4.14.1",
```
Comment (jacksonjp) [inline package.json:20]: There are some breaking changes with node-sass 6, we can test it really well before merging

Comment (jacksonjp) [reply package.json:20]: Lets try and use sass as some are facing issues with node-sass compatibility with node versions

Comment (yukinoda) [reply package.json:20]: @jacksonjp 
Thanks for working on this 😄 
3e6248f

**Review by geekonthemove — APPROVED** [review]
Looks good 👍

**Pull request merged**